### PR TITLE
Add flake8 pre-commit hook and manually edit files to fix flake8 viol…

### PR DESCRIPTION
**Pull request by dan98765**
…ations

Mostly removing imports that never get used, multiple tests with the same name in a file, indentation, variables named `l`...

### CODE REVIEW

**Comment by dan98765** [comment]
Master has changed drastically since I first did this. I'll close this out and reopen after I get a chance to redo it.

**Comment by dan98765** [comment]
This is resurrected and updated.

Ignoring F401 (unused imports) across the repository since there are things imported for mypy that look unused to flake8. 

@syrusakbary this is ready for your 👀

File: graphql/type/tests/test_validation.py [hunk 1]
```diff
@@ -1558,26 +1567,6 @@ def test_rejects_an_object_with_an_incorrectly_typed_interface_argument(self):
...
-        )
-
-        with raises(AssertionError) as excinfo:
-            schema_with_field_type(AnotherObject)
-
-        assert (
-            str(excinfo.value) == 'AnotherInterface.field expects type "String" '
-            'but AnotherObject.field provides type "SomeScalar".'
-        )
-
```
Comment (syrusakbary) [inline graphql/type/tests/test_validation.py:1580]: Why this is removed?

Comment (dan98765) [reply graphql/type/tests/test_validation.py:1580]: Hmm I think this was not meant to be removed. Probably an accident while resolving a merge conflict. Adding it back in.

Comment (dan98765) [reply graphql/type/tests/test_validation.py:1580]: Ohh wait nevermind; if you look at the test method directly beneath this you'll see its also called "test_rejects_an_object_with_a_differently_typed_Interface_field" and is the same; that's why I removed it.

**Comment by dan98765** [comment]
Closing in favor of https://github.com/graphql-python/graphql-core/pull/232

**Pull request closed without merging**